Sources/qiskit/datastructures/Matrix.swift

```diff
@@ -443,6 +457,74 @@ extension Matrix where T == Complex {
         return m
     }
 
+    public func eigh() throws -> (Vector<Double>, Matrix<Complex>) {
+        guard isHermitian else {
+            throw ArrayError.matrixIsNotHermitian
+        }
+
+        var jobz = Int8(86) // V: Compute eigenvalues and eigenvectors
+        var uplo = Int8(76) // L: Lower triangular part
+
+        var n = Int32(rowCount)
+        var lda = Int32(rowCount)
+        var info = Int32()
+
+        var w = Array(repeating: Double(), count: rowCount)
+
+        #if os(OSX) || os(iOS)
+
+            var a = flattenCol().map { __CLPK_doublecomplex(r: $0.real, i: $0.imag) }
```

Comment:
Notice that this line (and a few other below) increases the memory footprint of this method. Here, I am allocating double the size of the current matrix.